pkg/resourcecollector/resourcetransformation.go

```diff
@@ -9,10 +10,72 @@ import (
 	storkops "github.com/portworx/sched-ops/k8s/stork"
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+func GetGVK(Data []byte) (schema.GroupVersionKind, error) {
+	metadata := []metav1.TypeMeta{}
+	if err := json.Unmarshal(Data, &metadata); err != nil {
+		return schema.GroupVersionKind{}, err
+	}
+	gVersion, err := schema.ParseGroupVersion(metadata[0].APIVersion)
+	if err != nil {
+		return schema.GroupVersionKind{}, err
+	}
+	return gVersion.WithKind(metadata[0].Kind), nil
+}
+
+func IsObjectLabelsMatched(content map[string]interface{}, selector map[string]string) bool {
+	logrus.Info("Starting IsObjectLabelsMatched")
+	objectLabels, _, err := unstructured.NestedStringMap(content, "metadata", "labels")
+	if err != nil {
+		logrus.Error("error finding the metadata.labels from the unstructed content: ", content)
+	}
+	logrus.Infof("ObjectLabels %v, selector %v,output %v", objectLabels, selector, labels.SelectorFromSet(selector).Matches(labels.Set(objectLabels)))
+	return labels.SelectorFromSet(selector).Matches(labels.Set(objectLabels))
+	// labels.Equals(objectLabels, selector)
+}
+
+// GetGVKToTransformSpecMapper {GVK: []TransformSpecs{}}
+func GetGVKToTransformSpecMapper(transformName string, namespace string) (map[schema.GroupVersionKind][]stork_api.TransformSpecs, error) {
+	logrus.Info("Starting GetGVKToTransformSpecMapper")
+	gvkResourceTransform := make(map[schema.GroupVersionKind][]stork_api.TransformSpecs)
+	if transformName == "" || namespace == "" {
+		logrus.Errorf("empty name received for resource transformation/namespace")
+		return gvkResourceTransform, fmt.Errorf("empty name received for resource transformation/namespace")
+	}
+
+	resp, err := storkops.Instance().GetResourceTransformation(transformName, namespace)
```

Comment:
As this is called n times from
verifyNamespace - download namespace.json
VolumeStage - download storageclass.json, resources.json, crd.json
ApplicationStage - download resources.json, crd.json

Is good to have a cache for RT CR object instead of reaching out to apiserver

Reply:
We need to check also when the RT template reaches its high watermark interms of rules how big this in-memory copy shoots up. Else in case of multiple restore with multiple RT template this in memory size will bloat the whole cluster memory consumption